From 30dbacbbec79d5a35f457d56347e252f420d08c7 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:40:36 -0400
Subject: [ethernet] i40e: Enable l2tsel bit for VLAN tag control

Message-id: <1409658042-22689-276-git-send-email-sassmann@redhat.com>
Patchwork-id: 90474
O-Subject: [RHEL7.1 PATCH 275/281] i40e: Enable l2tsel bit for VLAN tag control
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shannon Nelson <shannon.nelson@intel.com>

Enable the l2tsel bit on Rx queue contexts that are assigned to VFs so
that the VF can get the stripped VLAN tag.

Change-ID: I7d9bc56238a9ea9baf5e8a97e69b9e27ebb9d169
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Signed-off-by: Ashish Shah <ashish.n.shah@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit c1d11cef92199764a3b0f35efc4a3583afedaddc)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c b/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c
index a81390c..339f69d 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c
@@ -350,6 +350,7 @@ static int i40e_config_vsi_rx_queue(struct i40e_vf *vf, u16 vsi_idx,
  rx_ctx.lrxqthresh = 2;
  rx_ctx.crcstrip = 1;
  rx_ctx.prefena = 1;
+ rx_ctx.l2tsel = 1;
 
  /* clear the context in the HMC */
  ret = i40e_clear_lan_rx_queue_context(hw, pf_queue_id);
-- 
1.7.1